core/src/test/java/org/infinispan/xsite/irac/IracTombstoneCleanupTest.java

```diff
@@ -352,18 +391,27 @@ void stopRecording() {
          recording = false;
          synchronized (this) {
             commandList.clear();
+            requests.clear();
+            forbidden.clear();
          }
 
       }
 
       @Override
       protected <T> CompletionStage<T> performRequest(Collection<Address> targets, ReplicableCommand command, ResponseCollector<T> collector, Function<ResponseCollector<T>, CompletionStage<T>> invoker, RpcOptions rpcOptions) {
+         CompletionStage<T> cs = super.performRequest(targets, command, collector, invoker, rpcOptions);
+
          if (recording && command instanceof CacheRpcCommand) {
             synchronized (this) {
+               if (forbidden.contains(command.getClass())) {
+                  throw CompletableFutures.asCompletionException(
+                        new IllegalStateException("Commands of " + command.getClass() + " are forbidden"));
+               }
                commandList.add((CacheRpcCommand) command);
+               requests.put(command.getCommandId(), cs);
```

Comment:
I probably would store the command as key instead of `command.getCommandId()`.

Reply:
I was going with that at first, but I don't know why the CI kept consistently failing/timing out and only worked when I started to use the id. I did not investigate it to see if it was just my luck or not.